File: src/panel/Keyboard.java
package panel;
import javafx.scene.canvas.Canvas;
import javafx.scene.canvas.GraphicsContext;
import javafx.scene.paint.Color;

final public class Keyboard extends Canvas {
    private GraphicsContext _gc;
    private short _height = 180;

    public Keyboard() {
        super();
        setHeight(185);
        setWidth(720);
        _gc = getGraphicsContext2D();
        for (short i = 0; i <= 24; i++) {
            renderNote(i , false);
        }

    }

    public void renderNote(short i, boolean isActive) {
        _gc.setFill(Color.GRAY);
        _gc.fillRect(i * 30, 0, 30 , 185);
        short h = (short) (isActive ? (_height - 20) : _height);
        short keyId = (short)( (i + 1) % 12);
        switch (keyId) {
            case 2:
            case 4:
            case 7:
            case 9:
            case 11:
                _gc.setFill(Color.BLACK);
                _gc.fillRect(i * 30, 0, 30, h);
                break;
            default:
                _gc.setFill(Color.WHITE);
                _gc.fillRect(i * 30 + 1, 0, 30 - 1, h);
                break;
        }
    }
}

File: src/panel/Keyboard.kt
package panel

import javafx.scene.canvas.Canvas
import javafx.scene.canvas.GraphicsContext
import javafx.scene.paint.Color

class Keyboard : Canvas() {
    private val _gc: GraphicsContext
    private val _height: Int = 180

    init {
        height = 185.0
        width = 720.0
        _gc = graphicsContext2D
        for (i in 0..24) {
            renderNote(i, false)
        }

    }

    fun renderNote(i: Int, isActive: Boolean) {
        _gc.fill = Color.GRAY
        _gc.fillRect((i * 30).toDouble(), 0.0, 30.0, 185.0)
        var h = _height;
        if(isActive){
            h -= 20
        }
        val keyId = ((i + 1) % 12)

        when (keyId) {
            2, 4, 7, 9, 11 -> {
                _gc.fill = Color.BLACK
                _gc.fillRect((i * 30).toDouble(), 0.0, 30.0, h.toDouble())
            }
            else -> {
                _gc.fill = Color.WHITE
                _gc.fillRect((i * 30 + 1).toDouble(), 0.0, (30 - 1).toDouble(), h.toDouble())
            }
        }
    }
}
